src/test/java/net/snowflake/ingest/streaming/internal/DataValidationUtilTest.java

```diff
@@ -313,29 +316,32 @@ public void testValidateAndParseBigDecimal() {
 
   @Test
   public void testValidateAndParseString() {
-    Assert.assertEquals(
-        "honk", DataValidationUtil.validateAndParseString("honk", Optional.empty()));
+    assertEquals("honk", validateAndParseString("honk", Optional.empty()));
 
     // Check max String length
     StringBuilder longBuilder = new StringBuilder();
-    for (int i = 0; i < DataValidationUtil.MAX_STRING_LENGTH + 1; i++) {
-      longBuilder.append("a");
+    for (int i = 0; i < 16 * 1024 * 1024; i++) {
+      longBuilder.append("č"); // max string length is measured in chars, not bytes
     }
-    String tooLong = longBuilder.toString();
+    String maxString = longBuilder.toString();
+    Assert.assertEquals(maxString, validateAndParseString(maxString, Optional.empty()));
 
-    try {
-      DataValidationUtil.validateAndParseString(tooLong, Optional.empty());
-      Assert.fail("Expected error for String too long");
-    } catch (SFException e) {
-      Assert.assertEquals(ErrorCode.INVALID_ROW.getMessageCode(), e.getVendorCode());
-    }
+    // max length + 1 should fail
```

Comment:
do we have a test verify max length-1 will succeed?

Reply:
Test added